include/small/Conv2DLayer.hpp

```diff
@@ -12,12 +12,170 @@
 
 #pragma once
 
+#include <cmath>
 #include <small.h>
 #include <small/buffers.hpp>
 #include <small/Layer.hpp>
 
 namespace small
 {
+namespace detail
+{
+    //************************************************************************
+    template <class BufferT>
+    void initialize_conv2d_buffers(
+        uint32_t          num_output_channels,
+        uint32_t          num_input_channels,
+        uint32_t          m_kernel_height,
+        uint32_t          m_kernel_width,
+        BufferT    const &filters,
+        BufferT    const &bias,
+        BufferT    const &bn_weight,            // gamma
+        BufferT    const &bn_bias,              // beta
+        BufferT    const &bn_running_mean,      // mu_hat
+        BufferT    const &bn_running_variance,  // sigma_hat^2
+        float      const &bn_eps,               // float?
+        bool              buffers_are_packed,
+        BufferT          &m_packed_filters,
+        BufferT          &m_packed_bias)
+    {
+        // ============ Filter weights ===========
+        if (filters.size() <
+            num_output_channels*num_input_channels*
+            m_kernel_height*m_kernel_width)
+        {
+            throw std::invalid_argument(
+                "*Conv2DLayer::ctor ERROR: "
+                "filters buffer too small.");
+        }
+
+        if (!buffers_are_packed)
+        {
+            // Pack the filter buffers for SMaLL use
+            small::pack_buffer(filters,
+                               FILTER_CONV,
+                               num_output_channels, num_input_channels,
+                               m_kernel_height, m_kernel_width,
+                               C_ib, C_ob,
+                               m_packed_filters);
+        }
+        else
+        {
+            std::copy(filters.data(),
+                      filters.data() + m_packed_filters.size(),
+                      m_packed_filters.data());
+        }
+
+        // ============ Bias term ===========
+        if (bias.size() > 0)
+        {
+            if (bias.size() != num_output_channels)
+            {
+                throw std::invalid_argument(
+                    "*Conv2DLayer::ctor ERROR: "
+                    "bias buffer incorrect size.");
+            }
+
+            // if (!buffers_are_packed)
+            BufferT packed_bias(num_output_channels);
+            std::copy(bias.data(),
+                      bias.data() + num_output_channels,
+                      packed_bias.data());
+            m_packed_bias = std::move(packed_bias);
```

Comment:
m_packed_bias was constructed with zero bytes (implicit in default construction).  Buffer does not have a resize method.  This is by design so that allocations like this are more explicit (make sure one REALLY wants to do this).  Since I do not have a way to expand the size of the buffer directly, the next cheapest thing is to construct the correct sized buffer and move assign to the member.